paradox-nakadi-consumer-boot/src/main/java/de/zalando/paradox/nakadi/consumer/boot/components/EventReceiverRegistry.java

```diff
@@ -214,24 +215,24 @@ public void stop() throws InterruptedException {
             }
         });
 
-        receiverMap.values().forEach(receiver -> {
+        // wait for ZK background tasks
+        Thread.sleep(2000);
```

Comment:
The code before my changes was waiting 2000 ms as well. I only moved the line up. This is a hotfix and I don't see an easy way of getting rid of the Thread.sleep.